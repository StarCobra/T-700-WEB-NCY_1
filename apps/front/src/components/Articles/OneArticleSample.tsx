import * as React from 'react';
import '../../style/article.scss'
import ArticleThumbnail from './ArticleThumbnail';
import ArticleTitleSample from './ArticleTitleSample';
import ArticleSampleContent from './ArticleSampleContent';
import ArticleDate from './ArticleDate';

export default function OneArticleSample() {
  return (
    <div className="articleSample">
            <ArticleThumbnail />
            <div className="articleSampleContent">
                <ArticleTitleSample />
                <ArticleSampleContent />
                <ArticleDate />
            </div>
        </div>
  )
}
 