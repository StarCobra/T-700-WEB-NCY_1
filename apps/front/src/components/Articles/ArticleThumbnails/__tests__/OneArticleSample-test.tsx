import React from "react";
import { act, create } from "react-test-renderer";
import OneArticleSample from "../OneArticleSample";

describe("Components > ArticleThumbnails > OneArticleSample", () => {
  const tests = [
    {
      description: "generic",
      props: {},
    },
  ];
  tests.forEach((test) => {
    it(`should render the component ${test.description}`, async () => {
      let root;
      await act(async () => {
        root = create(<OneArticleSample {...test.props} />);
      });
      expect(root).toMatchSnapshot();
    });
  });
});
