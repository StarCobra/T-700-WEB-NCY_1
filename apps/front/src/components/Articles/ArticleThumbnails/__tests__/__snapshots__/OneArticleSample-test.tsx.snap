// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > ArticleThumbnails > OneArticleSample should render the component generic 1`] = `
<div
  className="articleSample"
>
  <div
    className="articleThumbnail"
  >
    <img
      src="default"
    />
  </div>
  <div
    className="articleSampleContent"
  >
    <div
      className="articleTitle"
    >
      Bitcoin Collapse
    </div>
    <div
      className="articleSampleContent"
    >
      <p>
        In recent times, the once-revered and volatile king of cryptocurrencies, Bitcoin, has experienced a notable decline, leaving investors and enthusiasts puzzled. This downturn raises questions about the long-term sustainability of the digital currency and its role in the ever-evolving landscape of finance and technology. 
        <a>
          Plus ....
        </a>
      </p>
    </div>
    <div
      className="articleDateSample"
    >
      November 13, 2023
    </div>
  </div>
</div>
`;
